File: hdcd_source_spa_react_ts_springboot_uuppsd (2)/hdcd_source_spa_react_ts_springboot/ch0708/src/modules/board.ts
import { createReducer } from "typesafe-actions";
import { Board } from "../App"
import { createAction } from "redux-actions";

const FETCH = "board/FETCH";
const FETCH_SUCCESS = "board/FETCH_SUCCESS";
const FETCH_FAILURE = "board/FETCH_FAILURE";

export const fetchStart = createAction(FETCH);
export const fetchSuccess = createAction(FETCH_SUCCESS, (data: string) => data);
export const fetchFailure = createAction(FETCH_FAILURE, (err: any) => err);

export interface BoardState {
  loading: { FETCH: boolean, FETCH_LIST: boolean};
  board: Board;
  boards: Board[];
  error: any;
}

const initialState: BoardState = {
  loading: {
    FETCH: false,
    FETCH_LIST: false,
  },
  board: { boardNo: '', title: '', writer: '', content: '', regDate: '' },
  boards: [],
  error: null,
};

const board = createReducer(
  initialState,
  {
    [FETCH]: (state) => ({
      ...state,
      loading: {
        ...state.loading,
        FETCH: true,
      },
    }),
    [FETCH_SUCCESS]: (state, action) => ({
      ...state,
      loading: {
        ...state.loading,
        FETCH: false,
      },
      board: action.payload,
    }),
    [FETCH_FAILURE]: (state) => ({
      ...state,
      loading: {
        ...state.loading,
        FETCH: false,
      },
    }),
  },
);

export default board;

File: hdcd_source_spa_react_ts_springboot_uuppsd (2)/hdcd_source_spa_react_ts_springboot/ch1309/src/main/java/org/hdcd/domain/result/Board.java
package org.hdcd.domain.result;

import java.time.LocalDateTime;

import lombok.ToString;

@ToString
public class Board {

	private int boardNo;
	private String title;
	private String content;
	private String writer;
	private LocalDateTime regDate;
	
	public Board(int boardNo, String title, String content, String writer, LocalDateTime regDate) {
		super();
		this.boardNo = boardNo;
		this.title = title;
		this.content = content;
		this.writer = writer;
		this.regDate = regDate;
	}
	
}
